### PR TITLE
Bare Metal support - initial commit

### PR DESCRIPTION
**Pull request by fgelcer**
* few things still must be done (TODO):
** create specific nemesis group
** cleanup loaders
** cleanup monitor

### CODE REVIEW

**Comment by fgelcer** [comment]
i have run on GCE as well (as it uses the same code as the one changed for bare metal), and it worked as a charm.
the code for bare metal has being failing, so couldn't have even 1 successful execution so far (opened a bug on scylla-entreprise - [#843](https://github.com/scylladb/scylla-enterprise/issues/843)

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1382,7 +1405,7 @@ def detect_disks(self, nvme=True):
         :param nvme: NVMe(True) or SCSI(False) disk
         :return: list of disk names
         """
-        patt = ('nvme0n*', 'nvme0n\w+') if nvme else ('sd[b-z]', 'sd\w+')
+        patt = ('nvme*n*', 'nvme\w+n\w+') if nvme else ('sd[b-z]', 'sd\w+')
```
Comment (bentsi) [inline sdcm/cluster.py]: what are patt[0] and patt[1], why do we need them? please separate to different variables, smth. like:
device_pattern, disk_pattern = ('nvme*n*', 'nvme\w+n\w+') if nvme else ('sd[b-z]', 'sd\w+')

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1395,7 +1418,8 @@ def scylla_setup(self, disks):
         Setup scylla
         :param disks: list of disk names
         """
-        result = self.remoter.run('/sbin/ip -o link show |grep ether |awk -F": " \'{print $2}\'', verbose=True)
+        result = self.remoter.run('/sbin/ip -o address show | grep -v inet6 | grep -v "lo " | awk -F" " \'{print $2}\'',
```
Comment (bentsi) [inline sdcm/cluster.py]: `better: `/sbin/ip -f inet -o address show | grep -v "lo " | awk -F" " \'{print $2}\''`

File: sdcm/cluster_baremetal.py [hunk 3]
```diff
@@ -16,9 +16,9 @@ class NodeIpsNotConfiguredError(Exception):
 
 class PhysicalMachineNode(cluster.BaseNode):
 
-    def __init__(self, name, public_ip, private_ip, credentials, base_logdir=None, node_prefix=None):
+    def __init__(self, name, public_ip, private_ip, credentials, base_logdir=None, node_prefix=None, baremetal_username=None):
         ssh_login_info = {'hostname': None,
-                          'user': credentials.name,
+                          'user': baremetal_username if baremetal_username else credentials.name,
```
Comment (bentsi) [inline sdcm/cluster_baremetal.py]: you need to use standard credentials, no need to create specific user creds for Physical cluster

File: sdcm/cluster.py [hunk 4]
```diff
@@ -1315,12 +1325,26 @@ def clean_scylla(self):
...
+        scylla_mount = self.remoter.run('mount | grep scylla | awk -F" " \'{print $3}\'')
+        if scylla_mount.stdout:
+            self.remoter.run('sudo umount %s' % scylla_mount.stdout.strip())
+
+        md_device = self.remoter.run('ls /dev/md*', ignore_status=True)
+        if md_device.return_code == 0:
+            for dev in md_device.stdout.splitlines():
+                self.remoter.run('sudo mdadm --stop %s' % dev.split('/')[-1])
+
+        self.remoter.run('sudo pkill -9 node_exporter', ignore_status=True)
```
Comment (bentsi) [inline sdcm/cluster.py]: better to run multiple commands as a script, see examples in code

File: test-cases/longevity/longevity-bare-metal.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+             "cassandra-stress counter_write cl=QUORUM duration=240m -schema 'replication(factor=3) compaction(strategy=DateTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=5 -pop seq=1..20971520"
+             ]
+
+n_db_nodes: 3
+n_loaders: 3
+n_monitor_nodes: 1
+ip_ssh_connections: 'private'
+
+# all the below are lists of physical machine IPs
+db_nodes_public_ip: [10.0.4.131, 10.0.4.132, 10.0.4.133]
```
Comment (fruch) [inline test-cases/longevity/longevity-bare-metal.yaml]: I would suggest putting the Sahara configuration into a separate config file
```bash
SCT_CONFIG_FILES='["test-cases/longevity/longevity-3months.yaml", "setups/sahara-bare-metal.yaml"]'
```

File: sdcm/cluster.py [hunk 6]
```diff
@@ -165,6 +165,8 @@ class Setup(object):
     REUSE_CLUSTER = False
     MIXED_CLUSTER = False
     MULTI_REGION = False
+    # TODO: cleanup defaults to True or False?
```
Comment (fruch) [inline sdcm/cluster.py]: I think it better default to `False` to begin with

**Comment by bentsi** [comment]
@fgelcer what is the status of this?

**Comment by fgelcer** [comment]
@bentsi , not ready, as it is still flaky... juggling with this task while running other ones

**Comment by bentsi** [comment]
any progress here?

**Comment by fgelcer** [comment]
> any progress here?

its priority was lowered, and no progress was made here since last update.

**Pull request closed without merging**